Apps/W1/Shopify/app/src/Products/Codeunits/ShpfyUpdateItem.Codeunit.al

```diff
@@ -97,11 +99,17 @@ codeunit 30188 "Shpfy Update Item"
     /// <param name="ItemVariant">Parameter of type Record "Item Variant".</param>
     /// <returns>Return value of type Boolean.</returns>
     local procedure DoUpdateItemVariant(ShopifyVariant: Record "Shpfy Variant"; var ItemVariant: Record "Item Variant"): Boolean
+    var
+        IsModified: Boolean;
     begin
         if ItemVariant.Description <> ShopifyVariant.Title then begin
             ItemVariant.Description := ShopifyVariant.Title;
-            exit(ItemVariant.Modify());
+            IsModified := true;
         end;
+        ProductEvents.OnDoUpdateItemVariantBeforeModify(Shop, ShopifyVariant, ItemVariant, IsModified);
+        if IsModified then
+            exit(ItemVariant.Modify());
+        exit(false);
```

Comment:
I am not the biggest fan of this change. This event can override IsModified, and don't sync the item even though it has changed.

I would prefer something like this
 
ProductEvents.OnDoUpdateItemBeforeModify(Shop, ShopifyProduct, Item, IsModifiedByEvent);
        if IsModified or IsModifiedByEvent then
            exit(Item.Modify());
        exit(false);